### PR TITLE
fix(results): fix results tables when missing data

### PR DESCRIPTION
**Pull request by soyacz**
In case some columns/rows are missing table breaks (wrong data in columns or looks just bad).

Fix by revamping the way we pass data to renderer: transform it to more organized data structure with known columns and rows that are present and properly handling missing values.

fixes: https://github.com/scylladb/argus/issues/417

### CODE REVIEW

**Comment by soyacz** [comment]
before fixing:
image

after fixing:
fixed-table

and example with missing single value:
Screenshot from 2024-07-26 09-33-59

File: frontend/TestRun/ResultsTab.svelte [hunk 1]
```diff
@@ -43,18 +100,15 @@
...
 
-    const getColumnType = (result, columnName) => {
-        return result.meta.columns_meta.find((col) => col.name === columnName).type;
-    };
-
     onMount(() => {
         fetchResults();
     });
 
     const styleMap = {
```
Comment (k0machi) [inline frontend/TestRun/ResultsTab.svelte:107]: Const values ideally should be uppercase split by underscore: `STYLE_MAP`. Also you can move this into `Common/TestStatus.js` and import it here, that's where a lot of those helper objects are.

File: frontend/TestRun/ResultsTab.svelte [hunk 2]
```diff
@@ -6,17 +6,74 @@
     export let test_id = "";
     let results = [];
 
+    const transform_to_table = (table_data) => {
```
Comment (k0machi) [inline frontend/TestRun/ResultsTab.svelte:9]: should be `transformToTable` here (and `tableData`)

**Review by k0machi — COMMENTED** [review]
LGTM, aside from minor style comments

File: frontend/TestRun/ResultsTab.svelte (hunk 2 above)
Comment (soyacz) [reply frontend/TestRun/ResultsTab.svelte:9]: python style 😆

File: frontend/TestRun/ResultsTab.svelte (hunk 1 above)
Comment (soyacz) [reply frontend/TestRun/ResultsTab.svelte]: fixed

File: frontend/TestRun/ResultsTab.svelte (hunk 2 above)
Comment (soyacz) [reply frontend/TestRun/ResultsTab.svelte]: fixed

**Review by k0machi — APPROVED** [review]

**Pull request merged**